### ISSUE TITLE
Issue connecting metamask-sdk v0.5.0 to React app.

### ISSUE THREAD

**Issue by snwlprdgthb** [MetaMask/metamask-sdk#209]
Errors occur when trying to include the latest version of metamask-sdk (v0.5.0) and get the provider object in create-react-app.
After I rolled back to metamask-sdk v0.4.2, everything works fine as before.

The following simplified code works fine with v0.4.2:

```
import { MetaMaskSDK } from '@metamask/sdk';
import { useState, useEffect } from "react";


const sdk = new MetaMaskSDK({
  enableDebug: true,
  autoConnect: {
    enable: true
  },
  dappMetadata: {
    url: window.location.href,
    name: document.title,
  },
  logging: {
    developerMode: true,
  },
  storage: {
    enabled: true,
  }
});

const ethereum = sdk.getProvider();

function App() {  
  const connect = () => {
    ethereum
      .request({
        method: "eth_requestAccounts",
        params: [],
      })
      .then((res) => console.log("request accounts", res))
      .catch((e) => console.log("request accounts ERR", e));
  };

  return (
    <div className="App">
        <button style={{ padding: 10, margin: 10 }} onClick={connect}>
          Testing the metamask/sdk connection
        </button>
    </div>
  );
}

export default App;

```

The following error logs occur when updating metamask-sdk to version v0.5.0 (this happens regardless of where the provider is initialized, be it useEffect or in the connect function):

<img width="779" alt="image" src="https://github.com/MetaMask/metamask-sdk/assets/37382958/b4f0c693-e0cf-46c4-b8db-575153fc7abc">

Library versions from package.json:

```
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
    "react-scripts": "5.0.1",
```

- Platform: ReactJS
- MetaMask SDK Version: 0.5.0
- Browser (if relevant): Latest Google Chrome

**Comment by christopherferreira9** [comment]
Hi @snwlprdgthb! With version 0.5.0 we introduced a new async init to the SDK to improve stability and support multiple providers.
This means that doing something like `const ethereum = sdk.getProvider()` might not always work.
Take a look at the [example](https://github.com/MetaMask/metamask-sdk/tree/main/packages/examples/create-react-app) we just updated that includes this new functionality 🚀
